### ISSUE TITLE
RTL font in query screen is not readable. 

### ISSUE THREAD

**Issue by rotexhawk** [sqlitebrowser/sqlitebrowser#884]
### Details for the issue

Here is a sample query that has both languages. In DB4S the english text after the RTL text get's placed on top of the RTL text. Please seen screenshot below. It is also very hard to write the RTL text because the cursor position is kind of backwards. I know it is a mac issue, but I know some apps have figured out a way around it.

Thank you for the great tool. 

SELECT *
FROM other where word like '%عشق%'
order by 
	case when word like 'عشق' then 1
	case when word like 'عشق' then 1

![screen shot 2016-11-26 at 6 55 37 pm](https://cloud.githubusercontent.com/assets/12015105/20644663/9d5d5220-b40a-11e6-948c-eb08d5f8e207.png)

#### I'm opening this issue because:

- [ ] DB4S is crashing
- [x] DB4S has a bug
- [ ] DB4S needs a feature
- [ ] DB4S has another problem

#### I'm using DB4S on:

- [ ] Windows: ( _version:_ ___ )
- [ ] Linux: ( _distro:_ ___ )
- [x] Mac OS: ( _version: 10.12.1 )
- [ ] Other: ___

#### I'm using DB4S version:

- [x] 3.9.1
- [ ] 3.9.0
- [ ] Other: ___

#### I have also:

- [ ] Tried out the nightly version: https://github.com/sqlitebrowser/sqlitebrowser#windows
- [x] Searched for an existing similar issue: https://github.com/sqlitebrowser/sqlitebrowser/issues?utf8=%E2%9C%93&q=is%3Aissue%20

**Comment by justinclift** [comment]
@SafaAlfulaij Any ideas? :smile:

**Comment by SafaAlfulaij** [comment]
```
SqlTextEdit::SqlTextEdit(QWidget* parent) :
    QsciScintilla(parent)
{
    // Create lexer object if not done yet
if(sqlLexer == 0)
```

I think its because of `QsciScintilla`. :\

**Comment by justinclift** [comment]
😦

**Comment by MKleusberg** [comment]
It's even documented [here](http://www.scintilla.org/ScintillaDoc.html):
```
Scintilla does not properly support right-to-left languages like Arabic and Hebrew. While text in these languages may appear correct, it is not possible to interact with this text as is normal with other editing components.
```
So it's probably not going to change any time soon :frowning_face:

**Comment by MKleusberg** [comment]
Bug report: https://sourceforge.net/p/scintilla/bugs/1590/
It's so ridiculous that they are not even interested in fixing this :frowning:

**Comment by justinclift** [comment]
The QScintilla approach to "Community" is pretty terrible too.  I've I'd known then what I knew now, I would have recommended we find an alternative.

But that's hindsight in action.  It's still better than what we had. :wink:

Maybe we should add a longer term goal of finding a replacement?

**Comment by clayheaton** [comment]
I'm having this problem, too, unfortunately. R-to-L is invoked when I paste Arabic between the `'%%'` percent signs and all following text breaks. Sadly, this is going to force me back to SQLite Manager in Firefox for the time being, since I'm on an Arabic heavy project.

![editor_arabic](https://user-images.githubusercontent.com/205364/29850423-9e2ca3d4-8cfb-11e7-9238-40bf301843f1.png)

```sql
SELECT * FROM files_sheets WHERE header_values LIKE '%اسم العمل الجراحي%'; 
```

One suggestion would be to allow users to turn off the fancy SQL editor and just use a plain text box there.

**Comment by justinclift** [comment]
@clayheaton Yeah, that might be workable.  @MKleusberg how feasible would it be to have a toggle for QScintilla, so it can be turned off by people who find it too broken?

As a data point, there's probably going to be a fair number of people looking for an alternative to SQLite Manager over the coming months.  With Firefox changing how plugins are required to operate, and SQLite Manager barely being maintained, it's unlikely to be rewritten to continue working.

**Comment by mgrojo** [comment]
I just read this recent update about the Scintilla bug: https://sourceforge.net/p/scintilla/bugs/1590/

> Hassan DRAGA - 2017-10-29
> 
> now fixed on latest version of Scintilla

**Comment by justinclift** [comment]
@mgrojo Wow, that's good news. :smile:

We should probably investigate updating our Scintilla/QScintilla then and see if RTL text works ok.  Hmmm, maybe in a temporary branch or something for experimentation purposes though to check first, just in case.

**Comment by justinclift** [comment]
It looks like the base [Scintilla](https://sourceforge.net/projects/scintilla/) fixed the RTL problem ages ago, but the [QScintilla library](https://www.riverbankcomputing.com/software/qscintilla/download) that builds on it (which we use) hasn't been updated in years, and therefore doesn't include the fixes.

Looking around online just now, I'm not seeing an updated fork. :frowning: 

Anyone know if there's an update version somewhere, or alternative we should probably be switching to?

**Comment by scottfurry** [comment]
Riverbank Computing [releases](https://www.riverbankcomputing.com/software/qscintilla/download) the Qt implementation that DB4S builds on. Looks like another library that should be regularly pulled into the tree.

**Comment by justinclift** [comment]
Unfortunately, it doesn't appear to have been updated in years.  It seems to be based upon Scintilla 2.x, whereas Scintilla 4.x is the current release series (with the RTL fixes).

**Comment by justinclift** [comment]
Hmmm, unless I'm just misunderstanding the QScintilla version numbers, and they don't correspond to the Scintilla ones?

**Comment by scottfurry** [comment]
@justinclift - There may be _slight_ confustion. I just dug into the change log for QScintilla and came up with this:
```
v2.11 10th February 2019
  - Based on Scintilla v3.10.1.
```
It looks like Riverbank is one major revision behind.

**Comment by justinclift** [comment]
Interesting.  Hopefully it gets updated to recent Scintilla 4.x at some point then, so we can have proper RTL support. :smile:

**Comment by Ykulvaarlck** [comment]
i am waiting hoping for this to get fixed, i am working with a RTL language and troubleshooting the DB is a massive pain still.
